src/StaticWebAssetsSdk/Targets/Microsoft.NET.Sdk.StaticWebAssets.Compression.targets

```diff
@@ -0,0 +1,184 @@
+      $(ResolvePublishStaticWebAssetsDependsOn);
+      _ResolveCompressedFilesForPublish;
+    </ResolvePublishStaticWebAssetsDependsOn>
+
+  </PropertyGroup>
+
+  <Target
+    Name="_ResolveCompressedFiles"
+    DependsOnTargets="$(ResolveCompressedFilesDependsOn)">
+
+    <ItemGroup>
+      <_CompressedFileIncludePatternForBuild
+        Include="@(CompressionConfiguration)"
+        Condition="'%(Stage)' == 'Build'" />
+    </ItemGroup>
+
+    <ResolveCompressedAssets
+      CandidateAssets="@(StaticWebAsset)"
+      CompressionConfigurations="@(_CompressedFileIncludePatternForBuild)"
+      ExplicitAssets="@(AssetToCompress)"
+      OutputBasePath="$(IntermediateOutputPath)">
+
+      <Output
+        TaskParameter="AssetsToCompress"
+        ItemName="_CompressedStaticWebAssets" />
+
+    </ResolveCompressedAssets>
+
+    <ItemGroup>
+      <StaticWebAsset Include="@(_CompressedStaticWebAssets)" />
+    </ItemGroup>
+
+  </Target>
+
+  <Target
+    Name="_ResolveCompressedFilesForPublish"
+    DependsOnTargets="$(ResolveCompressedFilesForPublishDependsOn)">
+
+    <ItemGroup>
+
+      <_CompressedFileIncludePatternForPublish
+        Include="@(CompressionConfiguration)"
+        Condition="'%(Stage)' == 'Publish'" />
+
+      <_CandidateAssetsForPublish
+        Include="@(StaticWebAsset)"
+        Condition="'%(AssetKind)' != 'Build'" />
+
+    </ItemGroup>
+
+    <ResolveCompressedAssets
+      CandidateAssets="@(_CandidateAssetsForPublish)"
+      CompressionConfigurations="@(_CompressedFileIncludePatternForPublish)"
+      ExplicitAssets="@(AssetToCompress)"
+      OutputBasePath="$(IntermediateOutputPath)">
+
+      <Output
+        TaskParameter="AssetsToCompress"
+        ItemName="_CompressedStaticWebAssetsForPublish" />
+
+    </ResolveCompressedAssets>
+
+    <ItemGroup>
+      <StaticWebAsset Include="@(_CompressedStaticWebAssetsForPublish->'%(FullPath)')" />
+      <FilesToCompress Remove="@(FilesToCompress)" />
+    </ItemGroup>
+
+  </Target>
+
+  <Target
+    Name="_CompressFiles"
+    DependsOnTargets="$(CompressFilesDependsOn)"
+    BeforeTargets="_SplitStaticWebAssetsByCopyOptions">
+
+    <ItemGroup>
+      <_GZipCompressedStaticWebAssets Include="@(_CompressedStaticWebAssets)" Condition="'%(AssetTraitName)' == 'Content-Encoding' and '%(AssetTraitValue)' == 'gzip'" />
+      <_BrotliCompressedStaticWebAssets Include="@(_CompressedStaticWebAssets)" Condition="'%(AssetTraitName)' == 'Content-Encoding' and '%(AssetTraitValue)' == 'br'" />
+    </ItemGroup>
+
+    <GZipCompress Condition="'@(_GZipCompressedStaticWebAssets)' != ''"
+      FilesToCompress="@(_GZipCompressedStaticWebAssets)" />
+
+    <PropertyGroup Condition="'$(DOTNET_HOST_PATH)' == ''">
+      <_DotNetHostDirectory>$(NetCoreRoot)</_DotNetHostDirectory>
+      <_DotNetHostFileName>dotnet</_DotNetHostFileName>
+      <_DotNetHostFileName Condition="'$(OS)' == 'Windows_NT'">dotnet.exe</_DotNetHostFileName>
+    </PropertyGroup>
```

Comment:
Not sure where this is defined, but it's duplicated across targets.

Can we try putting it on a global propertygroup (outside of the target) or factor it into a common target?